model/cicd/metrics.yaml

```diff
@@ -0,0 +1,81 @@
+groups:
+  - id: metric.cicd.pipeline.run.duration
+    type: metric
+    metric_name: cicd.pipeline.run.duration
+    brief: 'Duration of a pipeline run grouped by pipeline and result.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+      - ref: cicd.pipeline.result
+        requirement_level: required
+      - ref: error.type
+        requirement_level:
+          conditionally_required: If and only if the pipeline run failed.
+  - id: metric.cicd.pipeline.run.active
+    type: metric
+    metric_name: cicd.pipeline.run.active
+    brief: 'The number of pipeline runs currently executing.'
+    instrument: updowncounter
+    unit: "{run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: required
+  - id: metric.cicd.pipeline.run.time_in_queue
+    type: metric
+    metric_name: cicd.pipeline.run.time_in_queue
+    brief: 'The duration a pipeline run takes from being triggered to the start of execution.'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.pipeline.run.queued
+    type: metric
+    metric_name: cicd.pipeline.run.queued
+    brief: 'The number of pipeline runs waiting for their start of execution.'
+    instrument: updowncounter
+    unit: "{run}"
+    stability: experimental
+    attributes:
+      - ref: cicd.pipeline.name
+        requirement_level: recommended
+  - id: metric.cicd.worker.count
+    type: metric
+    metric_name: cicd.worker.count
+    brief: 'The number of workers on the CICD system by state.'
+    instrument: updowncounter
+    unit: "{count}"
+    stability: experimental
+    attributes:
+      - ref: cicd.worker.state
+        requirement_level: required
+  - id: metric.cicd.pipeline.run.errors
+    type: metric
+    metric_name: cicd.pipeline.run.errors
```

Comment:
Could it be derived from `metric.cicd.pipeline.run.duration`? It could be if there is just one error per run, but I assume it's not the case, right?

Reply:
There might be errors in a pipeline run that are non fatal, ie they are suppressed or in a parallel stage multiple stages could have an error.
Ie this error count might not be the same as the count of `metric.cicd.pipeline.run.duration` with run result `failure`.

We did think that this metric might either be derived as a span metric from a run trace or reported directly by the CICD system controller.

Reply:
I have added this as a note in 02e044a5f38bc91ce80a14d267e79ee16c7cb7ca.